lib/runtime/version.go

```diff
@@ -30,9 +30,24 @@ type Version struct {
 }
 
 var (
+	// Blake2-8("TaggedTransactionQueue")
+	encodedTaggedTransactionQueue = [8]byte{0xd2, 0xbc, 0x98, 0x97, 0xee, 0xd0, 0x8f, 0x15}
```

Comment:
nit it would be fancy to have a new function in `lib/common/hasher.go`:

```go
// MustBlake2b8 returns the 8-bit blake2b hash of the input data
// and panics on error.
func MustBlake2b8(data []byte) (digest [8]byte) {
	const bytes = 1
	hasher, err := blake2b.New(bytes, nil)
	if err != nil {
		panic(err)
	}

	_, err = hasher.Write(data)
	if err != nil {
		panic(err)
	}

	digestBytes := hasher.Sum(nil)
	copy(digest[:], digestBytes)
	return digest
}
```

and then use it here using

```suggestion
	encodedTaggedTransactionQueue = common.MustBlake2b8([]byte("TaggedTransactionQueue"))
```

Reply:
Really like the idea but didn't seem to be working for me. I feel like I remember its actually an encoding of the whole function signature, but I might be wrong. Perhaps @danforbes would know

Reply:
The API ID is the first 8 bytes of the Blake2b hash of the API name. Ref https://github.com/paritytech/substrate/blob/polkadot-v0.9.29/primitives/api/proc-macro/src/decl_runtime_apis.rs#L553-L562

Reply:
Nice!